src/components/settings-menu/BackupSection/BackupSection.tsx

```diff
@@ -0,0 +1,156 @@
+import lang from 'i18n-js';
+import React, { useCallback } from 'react';
+import { cloudPlatform } from '../../../utils/platform';
+import { ContactAvatar } from '../../contacts';
+import Menu from '../components/Menu';
+import MenuContainer from '../components/MenuContainer';
+import MenuItem from '../components/MenuItem';
+import { removeFirstEmojiFromString } from '@rainbow-me/helpers/emojiHandler';
+import WalletBackupTypes from '@rainbow-me/helpers/walletBackupTypes';
+import WalletTypes from '@rainbow-me/helpers/walletTypes';
+import { useManageCloudBackups, useWallets } from '@rainbow-me/hooks';
+import { useNavigation } from '@rainbow-me/navigation';
+import { abbreviations } from '@rainbow-me/utils';
+import { addressHashedEmoji } from '@rainbow-me/utils/profileUtils';
+
+const BackupSection = () => {
+  const { navigate } = useNavigation();
+  const { walletNames, wallets } = useWallets();
+  const { manageCloudBackups } = useManageCloudBackups();
+
+  const onPress = useCallback(
+    (walletId, name) => {
+      const wallet = wallets[walletId];
+      if (wallet.backedUp || wallet.imported) {
+        navigate('SettingsBackupView', {
+          imported: wallet.imported,
+          title: name,
+          type: 'AlreadyBackedUpView',
+          walletId,
+        });
+      } else {
+        navigate('SettingsBackupView', {
+          title: name,
+          type: 'NeedsBackupView',
+          walletId,
+        });
+      }
+    },
+    [navigate, wallets]
+  );
+
+  let cloudBackedUpWallets = 0;
+
+  return (
+    <MenuContainer>
+      <Menu>
+        {Object.keys(wallets)
+          .filter(key => wallets[key].type !== WalletTypes.readOnly)
+          .map(key => {
+            const wallet = wallets[key];
+            const visibleAccounts = wallet.addresses.filter(
+              (a: any) => a.visible
+            );
+            const account = visibleAccounts[0];
+            const totalAccounts = visibleAccounts.length;
+            const { color, label, address } = account;
+            if (wallet.backupType === WalletBackupTypes.cloud) {
+              cloudBackedUpWallets += 1;
+            }
+            let labelOrName = label;
+            if (!label) {
+              if (walletNames[address]) {
+                labelOrName = walletNames[address];
+              }
+            }
+
+            return (
```

Comment:
wondering if we should move this logic out of `return`, feels strange to have this logic (and logic below this comment) together with rendered components, wdyt?